pkg/nsx/services/ipblocksinfo/ipblocksinfo.go

```diff
@@ -0,0 +1,301 @@
+		}
+		// save private_tgw_ip_blocks path in set for profile associated with default project
+		if isDefault {
+			for _, privateTgwIpBlocks := range vpcConnectivityProfile.PrivateTgwIpBlocks {
+				privateTgwIPBlockPaths.Insert(privateTgwIpBlocks)
+			}
+		}
+	}
+
+	// get IPBlock CIDRs from NSX
+	if externalIPCIDRs, err = s.getIPBlockCIDRsByPaths(externalIPBlockPaths); err != nil {
+		return nil, nil, err
+	}
+	if privateTGWIPCIDRs, err = s.getIPBlockCIDRsByPaths(privateTgwIPBlockPaths); err != nil {
+		return nil, nil, err
+	}
+
+	return externalIPCIDRs, privateTGWIPCIDRs, nil
+}
+
+func isDefaultNetworkConfigCR(vpcConfigCR v1alpha1.VPCNetworkConfiguration) bool {
+	annos := vpcConfigCR.GetAnnotations()
+	val, exist := annos[common.AnnotationDefaultNetworkConfig]
+	if exist {
+		boolVar, err := strconv.ParseBool(val)
+		if err != nil {
+			log.Error(err, "failed to parse annotation to check default NetworkConfig", "Annotation", annos[common.AnnotationDefaultNetworkConfig])
+			return false
+		}
+		return boolVar
+	}
+	return false
+}
+
+func (s *IPBlocksInfoService) getIPBlockCIDRsByPaths(pathSet sets.Set[string]) ([]string, error) {
+	ipCIDRs := []string{}
+	for path := range pathSet {
+		ipblock, err := s.getIPBlockByPath(path)
+		if err != nil {
+			return nil, err
+		}
+		if ipblock.Cidr == nil {
+			return nil, fmt.Errorf("failed to get CIDR from ipblock %s", path)
+		}
+		log.V(2).Info("successfully get cidr for IPBblock", "path", path, "cidr", *ipblock.Cidr)
+		ipCIDRs = append(ipCIDRs, *ipblock.Cidr)
+	}
+	return ipCIDRs, nil
+}
+
+func (s *IPBlocksInfoService) getIPBlockByPath(path string) (*model.IpAddressBlock, error) {
+	// Examples of IPBlock path:
+	// /infra/ip-blocks/{ip-block-id}
+	// /orgs/{org-id}/projects/{project-id}/infra/ip-blocks/{ip-block-id}
+	var ipblock model.IpAddressBlock
+	var err error
+	ignoreIpblockUsage := true
+	pathSlice := strings.Split(path, "/")
+
+	if len(pathSlice) == 4 {
+		ipblock, err = s.NSXClient.IPBlockShortClient.Get(pathSlice[3], &ignoreIpblockUsage)
+	} else if len(pathSlice) == 8 {
+		ipblock, err = s.NSXClient.IPBlockClient.Get(pathSlice[2], pathSlice[4], pathSlice[7], &ignoreIpblockUsage)
+	} else {
+		return nil, fmt.Errorf("failed to parse IPBlock path '%s'", path)
+	}
+	if err != nil {
+		log.Error(err, "failed to get IPBlock CIDR from %s", path)
+		return nil, err
+	}
+	return &ipblock, nil
+}
+
+func (s *IPBlocksInfoService) getVPCConnectivityProfileByPath(path string) (*model.VpcConnectivityProfile, error) {
+	// An example of profile path: /orgs/{org-id}/projects/{project-id}/vpc-connectivity-profiles/{vpc-connectivity-profile-id}
+	pathSlice := strings.Split(path, "/")
+	if len(pathSlice) != 7 {
+		return nil, fmt.Errorf("failed to parse VPCConnectivityProfile path '%s'", path)
+	}
+	vpcConnectivityProfile, err := s.NSXClient.VPCConnectivityProfilesClient.Get(pathSlice[2], pathSlice[4], pathSlice[6])
+	return &vpcConnectivityProfile, err
+}
```

Comment:
I think we can put this function in pkg/nsx/services/vpc/vpc.go:
https://github.com/vmware-tanzu/nsx-operator/pull/704

Reply:
Updated, thanks!